.gitlab-ci.yml

```diff
@@ -84,7 +83,7 @@ stable-test:
   before_script:
     - mkdir -p ./artifacts/sccache/
   script:
-    - cargo +stable build --verbose --all-features
-    - RUST_BACKTRACE=1 cargo +stable test --workspace --verbose --no-default-features
-    - RUST_BACKTRACE=1 cargo +stable test --workspace --verbose --all-features
+    - cargo +stable build --verbose
+    - RUST_BACKTRACE=1 cargo +stable test --workspace --verbose
+    - cargo +stable build --release --features="dist-client,dist-server"
```

Comment:
This build job will barely reuse the cache from the previous one due to the different cargo profile. And will leave not reusable cache after itself, so the next job will take more time.

Reply:
How so? It's debug vs release anyways, so there is nothing shared.

Reply:
Checked, it doesn't reuse even 20% of the previous job cache. Let's put it to another CI job.

Reply:
artifacts are orthogonal, exactly, so there's not much sense in having this job here. It rather has to be as a prep job for the `dist-test` job

Reply:
which is not yet designed

Reply:
Cache usage is not much of concern of this PR, just to get the pipeline green so we can finally enforce it.